### ISSUE TITLE
ZStream.async no longer works for Scala.js client on lib versions after 2.0.10

### ISSUE THREAD

**Issue by KeithGrant** [zio/zio#8071]
The ZStream.async no longer streams the asynchronous callbacks from the node.js socket.on_data() function. No errors were reported. This worked for ZIO version 2.0.10 (Scals 3) but does not work for 11, 12 or 13. I tried ZIO.async for a one-off call. It worked for 10 but again, not 11 or 12.  Inserted print statement shows the socket is receiving data and calling the on_data function.

 val stream : ZStream[Any, Exception, Buffer] =
    ZStream.async { cb =>
      client.on_data(
        nodeStrings.data,
        (data: Buffer) =>
          println(s"cb: $data")
          cb(ZIO.succeed(Chunk(data)))
      )
    }

Note: client.on_data is a node.js 18.11.18 socket 'data' event

**Comment by jdegoes** [comment]
@KeithGrant Can you include a ZIO Test that reproduces the issue? Thank you! 🙏

**Comment by adamgfraser** [comment]
Yes would be good to have a reproducer for this. This completes normally for example:

```scala
test("async") {
  val stream =
    ZStream.async[Any, Nothing, Int] { cb =>
      scala.scalajs.js.timers.setInterval(1000) {
        cb(ZIO.succeed(Chunk(1)))
      }
    }
    for {
      _ <- stream.take(5).runDrain
    } yield assertCompletes
}
```

**Comment by KeithGrant** [comment]
I’ll look into creating a ZIO Test example but it may turn out to be a minimum client/server pair.

**Comment by adamgfraser** [comment]
Okay. If it is an issue with `ZStream.async` we should be able to reproduce it with a simpler asynchronous process.

**Comment by hearnadam** [comment]
@KeithGrant is this still an issue?

**Comment by KeithGrant** [comment]
I don’t know. I refactored the issue out of my code.
